pkg/server/billing.go

```diff
@@ -141,8 +139,54 @@ type BillingHandlerFunc func(
 ) ([]*billingpb.Record, int64)
 
 var handlers = map[string]BillingHandlerFunc{
-	"cpu": handleCPUBilling,
-	"ram": handleRAMBilling,
+	"cpu":       handleCPUBilling,
+	"ram":       handleRAMBilling,
+	"ip":        handleIPBilling,
+	"drive_ssd": handleSSDDriveBilling,
+	"drive_hdd": handleHDDDriveBilling,
+}
+
+var handleSSDDriveBilling BillingHandlerFunc = makeDriveBillingHandler("SSD")
+var handleHDDDriveBilling BillingHandlerFunc = makeDriveBillingHandler("HDD")
+
+func makeDriveBillingHandler(driveKind string) BillingHandlerFunc {
+	return func(log *zap.Logger, ltl LazyTimeline, i *ipb.Instance, vm LazyVM, res *billingpb.ResourceConf, last int64) ([]*billingpb.Record, int64) {
+		o, _ := vm()
+		storage := Lazy(func() float64 {
+			disks := o.Template.GetDisks()
+			total := 0.0
+			for _, disk := range disks {
+				capacity, _ := disk.GetFloat("SIZE")
+				driveType, _ := disk.GetStr("DRIVE_TYPE")
+
+				if driveType == driveKind {
+					total += capacity / 1000
+				}
+			}
+			return total
+		})
+
+		return handleCapacityBilling(log.Named("DRIVE"), storage, ltl, i, vm, res, last)
+	}
+}
+
+func handleIPBilling(log *zap.Logger, ltl LazyTimeline, i *ipb.Instance, vm LazyVM, res *billingpb.ResourceConf, last int64) ([]*billingpb.Record, int64) {
+	o, _ := vm()
+	ip := Lazy(func() float64 {
+		publicNetworks := 0.0
+		nics := o.Template.GetNICs()
+		for _, nic := range nics {
+			name, err := nic.GetStr("NETWORK")
+			if err != nil {
+				continue
+			}
+			if strings.Contains(name, "pub") {
```

Comment:
Type of network should be determined by ATTRIBUTE inside network, not the name(name can be changed)

https://github.com/slntopp/nocloud-driver-ione/blob/207e82e6c14cda17c1c2254c78818a6accb6e53e/pkg/driver/vnet.go#L75